### PR TITLE
Remove moment library from package.json

### LINKED ISSUE #365: replace moment library
As it is advised not to use moment anymore we need to find out how to replace its uses in our code.

### PR DESCRIPTION
**Pull request by MichaelGuldborg**
I replaced all moment imports with vanilla js functions
#365

### CODE REVIEW

**Comment by lucaslcode** [comment]
There's also `timeago.ts`

**Comment by MichaelGuldborg** [comment]
Aahh I missed timeago.ts

File: app/frontend/src/utils/timeAgo.ts [hunk 1]
```diff
@@ -1,9 +1,42 @@
...
+const yearMillis = 31557600000;
+const twoYearMillis = 31557600000 * 2;
+
+export function timeAgo(input: Date | string) {
+  if (input === undefined) return "";
+  const date = new Date(input);
+  const diffMillis = Date.now() - date.getTime();
 
-export function timeAgo(date: Date) {
   //the backend fuzzes times to 15 minute intervals
```
Comment (darrenvong) [inline app/frontend/src/utils/timeAgo.ts]: Would be good to have some unit tests for these since we are writing our own logic/ditching `moment`

Comment (andreaowu) [reply app/frontend/src/utils/timeAgo.ts]: +1

Comment (MichaelGuldborg) [reply app/frontend/src/utils/timeAgo.ts]: I've added a unit test to the timeago function

**Comment by lucaslcode** [comment]
There is a build error from the merge - I recently added another `fuzzed` parameter to `timeAgo` which defaults to false, which decides whether to do the < 15 minutes thing or not. Sorry!

File: app/frontend/src/utils/timeAgo.ts [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
-const dayMillis = 86400000;
-const twoDayMillis = 172800000;
-const weekMillis = 604800000;
-const twoWeekMillis = 1209600000;
-const monthMillis = weekMillis * 4;
-const twoMonthMillis = weekMillis * 8;
-const yearMillis = 31557600000;
-const twoYearMillis = 31557600000 * 2;
+export const minuteMillis = 60000;
+export const quaterHourMillis = 900000;
```
Comment (darrenvong) [inline app/frontend/src/utils/timeAgo.ts]:
```suggestion
export const quarterHourMillis = 900000;
```
Nitpick: spelling mistake

File: app/frontend/src/utils/timeAgo.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import {
+  dayMillis,
+  hourMillis,
+  minuteMillis,
+  monthMillis,
+  quaterHourMillis,
```
Comment (darrenvong) [inline app/frontend/src/utils/timeAgo.test.ts]:
```suggestion
  quarterHourMillis,
```

File: app/frontend/src/utils/timeAgo.test.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  twoMonthMillis,
+  twoWeekMillis,
+  twoYearMillis,
+  weekMillis,
+  yearMillis,
+} from "./timeAgo";
+
+const timeAgoMap = {
+  [minuteMillis]: "< 15 minutes ago",
+  [quaterHourMillis]: "15 minutes ago",
```
Comment (darrenvong) [inline app/frontend/src/utils/timeAgo.test.ts]:
```suggestion
  [quarterHourMillis]: "15 minutes ago",
```

File: app/frontend/src/utils/timeAgo.test.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  [yearMillis]: "1 year ago",
+  [twoYearMillis]: "2 years ago",
+};
+
+test("timeAgo function", () => {
+  const now = Date.now();
+
+  Object.keys(timeAgoMap).forEach((key: string) => {
+    const millis = parseInt(key);
+    const value = timeAgoMap[millis];
```
Comment (darrenvong) [inline app/frontend/src/utils/timeAgo.test.ts]:
```suggestion
    const expectedValue = timeAgoMap[millis];
```

File: app/frontend/src/utils/timeAgo.test.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    const date = new Date(now - millis);
+    const s = timeAgo(date);
+    expect(s).toBe(value);
```
Comment (darrenvong) [inline app/frontend/src/utils/timeAgo.test.ts]:
```suggestion
    const timeString = timeAgo(date);
    expect(timeString).toBe(expectedValue);
```
Probably both nitpicks as the tests are small enough you can guess the meaning, but I like to be explicit.

**Review by darrenvong — COMMENTED** [review]
Normally I wouldn't block an approval on these but since you have a build error, may as well fix that together ;)

**Comment by MichaelGuldborg** [comment]
Valid point @darrenvong
I've updated the PR

**Review by darrenvong — APPROVED** [review]

**Pull request merged**